sentry/src/main/java/io/sentry/SentryTransaction.java

```diff
@@ -0,0 +1,218 @@
+package io.sentry;
+
+import io.sentry.protocol.Contexts;
+import io.sentry.protocol.SentryId;
+import io.sentry.util.Objects;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class SentryTransaction extends SentryBaseEvent implements ISpan {
+  /** The transaction name. */
+  private @Nullable String transaction;
+
+  /** The moment in time when span was started. */
+  private final @NotNull Date startTimestamp;
+
+  /** The moment in time when span has ended. */
+  private @Nullable Date timestamp;
+
+  /** A list of spans within this transaction. Can be empty. */
+  private final @NotNull List<Span> spans = new ArrayList<>();
+  /**
+   * A hub this transaction is attached to. Marked as transient to be ignored during JSON
+   * serialization.
+   */
+  private @NotNull final transient IHub hub;
+
+  /** The {@code type} property is required in JSON payload sent to Sentry. */
+  @SuppressWarnings("UnusedVariable")
+  private @NotNull final String type = "transaction";
+
+  /** Creates transaction. */
+  SentryTransaction(final @NotNull String name) {
+    this(name, new SpanContext(), NoOpHub.getInstance());
```

Comment:
should we allow a NoOp transaction? `NoOpHub.getInstance()` makes me wonder

Reply:
This exist only because @bruno-garcia suggested that `NoOpHub` should not return nulls. I also don't like it. It's package private though, so at least does not leak.